zerver/lib/github.py

```diff
@@ -61,3 +66,44 @@ def get_latest_github_release_download_link_for_platform(platform: str) -> str:
         if verify_release_download_link(link):
             return link
     return "https://github.com/zulip/zulip-desktop/releases/latest"
+
+
+def get_issue_or_pr_cache_key(
+    owner: str, repo: str, type: str, issue_number: str
+) -> str:  # nocoverage
+    return f"github_issue_or_pr_url:{owner}/{repo}/{type}/{issue_number}"
+
+
+@cache_with_key(get_issue_or_pr_cache_key, timeout=3600)  # nocoverage
+def get_issue_or_pr_data(owner: str, repo: str, type: str, issue_number: str) -> Dict[str, str]:
+    # Change pull to pulls because in pull request url the type is pull but in api the type should be pulls.
+    if type == "pull":
+        type = "pulls"
+    api_url = f"https://api.github.com/repos/{owner}/{repo}/{type}/{issue_number}"
+    try:
+        response = GithubSession().get(api_url)
+        response.raise_for_status()
+        json_data: Dict[str, Any] = response.json()
```

Comment:
I think `json_data` is a misleading name for this variable -- it's just a dictionary at this point, not a JSON string, having been decoded from the original JSON.